File: app/src/main/java/org/amoustakos/boilerplate/injection/module/ActivityModule.java
package org.amoustakos.boilerplate.injection.module;

import org.amoustakos.boilerplate.injection.annotations.context.ActivityContext;
import org.amoustakos.boilerplate.ui.activities.BaseActivity;

import dagger.Module;
import dagger.Provides;

@Module
public class ActivityModule {
    private BaseActivity mActivity;

    public ActivityModule(BaseActivity activity) {
        mActivity = activity;
    }

    @Provides
    @ActivityContext
    BaseActivity providesContext() {
        return mActivity;
    }


}

File: app/src/main/java/org/amoustakos/boilerplate/injection/module/injectors/ActivityModule.kt
package org.amoustakos.boilerplate.injection.module.injectors

import dagger.Module
import dagger.Provides
import org.amoustakos.boilerplate.injection.annotations.context.ActivityContext
import org.amoustakos.boilerplate.ui.activities.BaseActivity

@Module
class ActivityModule(
		private val mActivity: BaseActivity
) {

	@Provides
	@ActivityContext
	internal fun providesContext(): BaseActivity = mActivity


}
